File: classes/parte-7-paradigmas/110-orientado-a-objetos-clases-objetos-y-estado/implementaciones/php/main.php
<?php
class Contador {
    public int $cuenta = 0;
    public function incrementar() {
        $this->cuenta++;
    }
}

$n = (int) trim(fgets(STDIN));
$c = new Contador();
for ($i = 0; $i < $n; $i++) {
    $c->incrementar();
}
echo "cuenta={$c->cuenta}\n";

File: classes/parte-7-paradigmas/110-orientado-a-objetos-clases-objetos-y-estado/implementaciones/rust/main.rs
use std::io::Read;

struct Contador {
    cuenta: i64,
}

impl Contador {
    fn nuevo() -> Self {
        Contador { cuenta: 0 }
    }
    fn incrementar(&mut self) {
        self.cuenta += 1;
    }
}

fn main() {
    let mut s = String::new();
    std::io::stdin().read_to_string(&mut s).unwrap();
    let n: i64 = s.trim().parse().unwrap();
    let mut c = Contador::nuevo();
    for _ in 0..n {
        c.incrementar();
    }
    println!("cuenta={}", c.cuenta);
}
